### PR TITLE
Update sdm machine

### PR DESCRIPTION
**Pull request by lievendoclo**

### CODE REVIEW

**Comment by lievendoclo** [comment]
Will need to check whether all new released dependency versions are compatible

**Comment by cdupuis** [comment]
@lievendoclo thanks for those changes. Great improvements.

I have some changes to suggest that I'll incorporate into this PR.

File: lib/machine/machine.ts [hunk 1]
```diff
@@ -15,26 +15,134 @@
...
 
 import {
+    AnyPush,
+    anySatisfied,
+    AutofixGoal, GitHubRepoRef,
+    goalContributors,
+    Goals, JustBuildGoal, onAnyPush, PushReactionGoal, ReviewGoal,
     SoftwareDeliveryMachine,
-    SoftwareDeliveryMachineConfiguration,
+    SoftwareDeliveryMachineConfiguration, StagingDeploymentGoal, StagingEndpointGoal, whenPushSatisfies,
```
Comment (cdupuis) [inline lib/machine/machine.ts]: Properly format imports.

File: lib/machine/machine.ts [hunk 2]
```diff
@@ -15,26 +15,134 @@
...
-        configuration,
-    });
-    addSpringSupport(sdm);
+    const sdm: SoftwareDeliveryMachine = createSoftwareDeliveryMachine(
+        {
+            name: "Spring software delivery machine",
+            configuration,
+        });
 
-    summarizeGoalsInGitHubStatus(sdm);
```
Comment (cdupuis) [inline lib/machine/machine.ts]: Add that back as it is useful and now doesn't break local.

File: lib/machine/machine.ts [hunk 3]
```diff
@@ -15,26 +15,134 @@
...
+    configureGoals(sdm);
+    configureGenerators(sdm);
+    configureExtensionPacks(sdm);
+    configureBuilder(sdm);
```
Comment (cdupuis) [inline lib/machine/machine.ts]: Please inline those as much as reasonable.

File: lib/machine/machine.ts [hunk 4]
```diff
@@ -15,26 +15,134 @@
...
+    } else {
+        configureDeploysForCloudSdm(sdm);
+    }
 
     return sdm;
 }
+
+export function configureDeploysForCloudSdm(sdm: SoftwareDeliveryMachine) {
+    configureLocalSpringBootDeploy(sdm);
+    sdm.addDeployRules(
```
Comment (cdupuis) [inline lib/machine/machine.ts]: addDeployRules is deprecated. Please replace with new approach.

File: lib/machine/machine.ts [hunk 5]
```diff
@@ -15,26 +15,134 @@
...
+            .deployTo(StagingDeploymentGoal, StagingEndpointGoal, StagingUndeploymentGoal)
+            .using(
+                {
+                    deployer: localExecutableJarDeployer(),
+                    targeter: ManagedDeploymentTargeter,
+                },
+            ),
+    );
+
+    sdm.addDisposalRules(
```
Comment (cdupuis) [inline lib/machine/machine.ts]: Completely remove disposal.

File: lib/machine/machine.ts [hunk 6]
```diff
@@ -15,26 +15,134 @@
...
+            description: "Create a new Java Spring Boot REST service",
+            paramsMaker: SpringProjectCreationParameters,
+            startingPoint: new GitHubRepoRef("spring-team", "spring-rest-seed"),
+            transform: [
+                ReplaceReadmeTitle,
+                SetAtomistTeamInApplicationYml,
+                TransformSeedToCustomProject,
+            ],
+        })
+        .addGeneratorCommand<SpringProjectCreationParameters>({
```
Comment (cdupuis) [inline lib/machine/machine.ts]: remove.

File: lib/machine/machine.ts [hunk 7]
```diff
@@ -15,26 +15,134 @@
...
+
+function configureExtensionPacks(sdm: SoftwareDeliveryMachine) {
+    sdm.addExtensionPacks(
+        SpringSupport,
+    );
+}
+
+export function configureBuilder(sdm: SoftwareDeliveryMachine) {
+    const mb = new MavenBuilder(sdm);
+    sdm.addBuildRules(
```
Comment (cdupuis) [inline lib/machine/machine.ts]: `addBuildRules` is deprecated too. Please register goal implementations directly.

**Comment by cdupuis** [comment]
We should also bring in the following things:
* Try to update spring boot
* A simple autofix

**Comment by cdupuis** [comment]
We should also rename this repo and seed to `spring-sdm-seed` to be in-line with other seeds.

File: lib/machine/machine.ts (hunk 6 above)
Comment (lievendoclo) [reply lib/machine/machine.ts]: done

File: lib/machine/machine.ts (hunk 7 above)
Comment (lievendoclo) [reply lib/machine/machine.ts]: Replaced with goal implementations

File: lib/machine/machine.ts (hunk 5 above)
Comment (lievendoclo) [reply lib/machine/machine.ts]: Removed

**Comment by lievendoclo** [comment]
Added autofix and code transform

**Pull request merged**